frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts

```diff
@@ -385,28 +321,170 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
     return metric;
   }
 
-  checkQueryNameError(queryName){
+  checkMetricKeyRequiredError(metricKeyMissing: Boolean){
+    if (metricKeyMissing) {
+      this.queryMetricKeyError.push(true);
+    }
+  }
+
+  checkQueryNameRequiredError(queryName: any){
     if (queryName == null || queryName === '') {
       this.queryNameError.push(true);
     }
   }
 
+  checkStatisticRequiredError(statistic: any) {
+    if (statistic == null || statistic === '') {
+      this.queryStatisticError.push(true);
+    }
+  }
+
+  checkComparisonStatisticRequiredError(compareFn: any, compareValue: any) {
+    if (compareFn == null || compareFn === '' || compareValue == null || compareValue === '') {
+      this.queryComparisonStatisticError.push(true);
+    }
+  }
+
+  ManageKeysControl(queryIndex: number, keyIndex: number) {
+    let keysArray;
+    if (queryIndex == null) {
```

Comment:
also seeing lots of double-equals used for comparison, usually it's technically fine but I think the advice of always using the strict triple-equals in javascript/typescript comes into play in examples like this spot.

Javascript does super weird stuff with double-equals, which one has to have it all memorized if you want to use double-equals. For instance in this spot, if what if `queryIndex` is `undefined`? Surprisingly, this would be true. What if it's `NaN` or empty string? It's false. Maybe that is fine for the code in this spot, but it's confusing. If you really are just checking for `null` and only `null`, you need `===`.

Reply:
At some places, I pass `null` as a param to ManageKeysControl to be able to take global queryIndex. But, just now observed that I can pass global queryIndex directly.